creator-node/src/contentAccess/contentAccessChecker.ts

```diff
@@ -0,0 +1,88 @@
+import { isRegisteredDiscoveryNode } from './helpers'
+import { CheckAccessArgs, CheckAccessResponse } from './types'
+import { recoverWallet, signatureHasExpired } from '../apiSigning'
+
+const PREMIUM_CONTENT_SIGNATURE_MAX_TTL_MS = 6 * 60 * 60 * 1000 // 6 hours
+
+/**
+ * Checks that all premium content headers are passed in.
+ * Checks that discovery node that generated signature is registered.
+ * Checks that signature is not too old.
+ * Checks that user requesting the content has access.
+ */
+export async function checkContentAccess({
+  cid,
+  contentAccessHeaders,
```

Comment:
what headers do you expect this string to have? instead of typing this as string can we do the JSON.parse outside this function and type this object with expected params?